### PR TITLE
docs/answers.rst: Add docs regarding answers

### PR DESCRIPTION
**Pull request by meetmangukiya**
# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: docs/answers.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+files.
+
+``answers.extraction.Extractor`` is the ``NodeVisitor`` that visits all the
+``section`` nodes. All the code text(``docutils.nodes.literal_block``) are
+extracted and stored separately. All the normal text(all nodes excluding
+``literal_block``) are extracted and stored separately. Filename, and section id
+is appended at the end of the text for backtracking the file from the text.
+
+.. note::
+   If the node has a nested section node, than the current node
```
Comment (Udayan12167) [inline docs/answers.rst]: `than` ->`then`

File: docs/answers.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Creating nodes and Edges
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For natural language processing ``spacy`` is used.
+
+Iterating over all the tokens:
+If `head<https://en.wikipedia.org/wiki/Head_(linguistics)>`_ of the token is
+the same as the token, then the token is ignored. If not, then
+`lemmatized<https://en.wikipedia.org/wiki/Lemmatisation>`_ form of token and
+head of token are created and an edge is formed between them.
```
Comment (Udayan12167) [inline docs/answers.rst]: Why is this done. Can you write to the best of your knowledge

File: docs/answers.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+``node.sections`` is a list of section ids.
+``node.code`` is a list of code of sections particular token appears in.
+``node.text`` is a list of text of sections in which the token appears.
+
+
+Scoring answers
+~~~~~~~~~~~~~~~
+
+A graph of question is formed the same way the graph of documentation is
+formed. Then one iterates over question graph's edges, and increments the score
```
Comment (Udayan12167) [inline docs/answers.rst]: Okay this is not clear at all.
1. Iterate over graph edges to get pairs of nodes.
2. Search for shortest path in the documentation graph for those pair of nodes.
3. Consider all nodes in the shortest path and the unique location where they come from.
4. Increment the count of these locations.
5. Return the location with the largest count/most relevancy. 

Note: Pleases add how count relates to relevancy explanation as well.

**Comment by Udayan12167** [comment]
As a more general comment. You should name elements as something unique so that you can refer to them later in the doc. Naming should be highlighted for quick correlation.

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/273/commits/d50d13cf1782e55ec45aef74e43a301093f78c37

**Comment by meetmangukiya** [comment]
reack 4478e67

**Comment by meetmangukiya** [comment]
@rultor merge

**Pull request merged**